extras/rya.prospector/src/main/java/org/apache/rya/prospector/mr/Prospector.java

```diff
@@ -0,0 +1,113 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.rya.prospector.mr;
+
+import static org.apache.rya.prospector.utils.ProspectorConstants.DEFAULT_VIS;
+import static org.apache.rya.prospector.utils.ProspectorConstants.EMPTY;
+import static org.apache.rya.prospector.utils.ProspectorConstants.METADATA;
+import static org.apache.rya.prospector.utils.ProspectorConstants.PERFORMANT;
+import static org.apache.rya.prospector.utils.ProspectorConstants.PROSPECT_TIME;
+import static org.apache.rya.prospector.utils.ProspectorUtils.connector;
+import static org.apache.rya.prospector.utils.ProspectorUtils.getReverseIndexDateTime;
+import static org.apache.rya.prospector.utils.ProspectorUtils.instance;
+import static org.apache.rya.prospector.utils.ProspectorUtils.writeMutations;
+
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.security.ColumnVisibility;
+import org.apache.commons.lang.time.DateUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.conf.Configured;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.util.Tool;
+import org.apache.hadoop.util.ToolRunner;
+import org.apache.rya.prospector.domain.IntermediateProspect;
+import org.apache.rya.prospector.utils.ProspectorUtils;
+
+/**
+ * Configures and runs the Hadoop Map Reduce job that executes the Prospector's work.
+ */
+public class Prospector extends Configured implements Tool {
+
+    private static long NOW = System.currentTimeMillis();
+
+    private Date truncatedDate;
+
+    public static void main(String[] args) throws Exception {
+        final int res = ToolRunner.run(new Prospector(), args);
+        System.exit(res);
+    }
+
+    @Override
+    public int run(String[] args) throws Exception {
+        final Configuration conf = getConf();
+
+        truncatedDate = DateUtils.truncate(new Date(NOW), Calendar.MINUTE);
+
+        final Path configurationPath = new Path(args[0]);
+        conf.addResource(configurationPath);
+
+        final String inTable = conf.get("prospector.intable");
+        final String outTable = conf.get("prospector.outtable");
+        final String auths_str = conf.get("prospector.auths");
+        assert inTable != null;
+        assert outTable != null;
+        assert auths_str != null;
+
+        final Job job = new Job(getConf(), this.getClass().getSimpleName() + "_" + System.currentTimeMillis());
```

Comment:
I'm not changing the behavior of the code, I'm just converting it from Groovy to Java, so I'm going to leave this as is.